### ISSUE TITLE
PasswordResetRequiredException does not trigger code email 

### ISSUE THREAD

**Issue by adilosa** [aws-amplify/amplify-ui#2488]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Create React App / Webpack 5

### What browsers are you seeing the problem on?

Firefox, Chrome

### Please describe your bug.

After importing users to a Cognito pool, the user is in the RESET_PASSWORD state [1]. When the user tries to sign in, they are kicked to the ConfirmPasswordReset screen and asked for a code, but a code email is never sent. 

This case is not currently handled in the UI.

If the user manually presses "Resend Code" the code email is immediately sent. 

[1] https://docs.aws.amazon.com/cognito/latest/developerguide/signing-up-users-in-your-app.html#signup-confirmation-verification-overview

### What's the expected behaviour?

A code email should be sent automatically.

### Help us reproduce the bug!

1. Import a user or otherwise get them in the "Reset Password" state
2. Go to the app and attempt to sign in with the username and any password
3. Arrive at the ConfirmPasswordReset screen with no code in your email inbox
4. Press "Resend Code" and see the code email received normally

example Cognito import CSV (may not match your user pool):
```
name,given_name,family_name,middle_name,nickname,preferred_username,profile,picture,website,email,email_verified,gender,birthdate,zoneinfo,locale,phone_number,phone_number_verified,address,updated_at,cognito:mfa_enabled,cognito:username
,,,,,,,,,test@example.com,True,,,,,,False,,,False,test@example.com
```

### Additional information and screenshots

Workaround:

I modified the `handleSignIn()` function to intercept the "PasswordResetRequiredException" and manually send the code using `Auth.forgotPassword()` (the same function the "Resend Code" button calls). This effectively manually presses "Resend Code" in this case without interrupting the UI flow. 

```jsx
<Authenticator
  services={{
    async handleSignIn({ username, password }) {
      return Auth.signIn(username, password).catch(error => {
        if (error.code === "PasswordResetRequiredException") {
          Auth.forgotPassword(username);
        }
        throw error;
      });
    },
  }}
  ...
/>
```

**Comment by slaymance** [comment]
@adilosa Thank you for your detailed bug report, we'll prioritize fixing this on our roadmap and update this issue accordingly.

**Comment by ErikCH** [comment]
Hi @adilosa !

I'm trying to reproduce this and I have not been able too. Is this still happening to you?

**Comment by ErikCH** [comment]
Hi @adilosa ! I was able to reproduce this issue. It should be out on our next release! Or you can install it and test it out using the @next tag `@aws-amplify/ui-react@next` !

**Comment by wlee221** [comment]
Hey @adilosa, this has been released in `@aws-amplify/ui-react@4.3.2`. Please reopen the issue if the issue persists after the upgrade, thanks!

**Comment by adilosa** [comment]
Hey @wlee221 and @ErikCH, I'm on `@aws-amplify/ui-react@4.3.3` now and can confirm it's working as expected without my workaround. Thanks so much!
